scimono-server/src/main/java/com/sap/scimono/api/Groups.java

```diff
@@ -92,25 +94,39 @@ public Response getGroup(@PathParam("id") @ValidId final String groupId) {
   }
 
   @GET
-  public Response getGroups(@DefaultValue(DEFAULT_START_INDEX) @QueryParam(START_INDEX_PARAM) int startIndex,
-      @DefaultValue(DEFAULT_COUNT) @QueryParam(COUNT_PARAM) int count, @QueryParam(START_ID_PARAM) @ValidStartId String startId,
+  public Response getGroups(@DefaultValue(DEFAULT_START_INDEX) @QueryParam(START_INDEX_PARAM) String startIndex,
+      @DefaultValue(DEFAULT_COUNT) @QueryParam(COUNT_PARAM) String count, @QueryParam(START_ID_PARAM) @ValidStartId String startId,
       @QueryParam(FILTER_PARAM) final String filter) {
     logger.trace("Reading groups with paging parameters startIndex {} startId {} count {}", startIndex, startId, count);
-    if (startIndex < 1) {
-      startIndex = 1;
+    int startIndexNum = 0;
+    int countNum = 0;
+    try {
+      startIndexNum = Integer.parseInt(startIndex);
+    } catch (NumberFormatException e) {
+      throw new InvalidInputException("StartIndex is not a numeric value or is out of range.");
```

Comment:
can be extracted in separate method or class indicating that it will validate this input param. I may be something like QueryParamParser validating and giving you the correct values